models/nlp/classification/huggingface/inference.py

```diff
@@ -0,0 +1,56 @@
+import zipfile
+
+import torch
+import valohai
+from transformers import AutoModelForSequenceClassification, AutoTokenizer
+
+from utils.serializers import get_serializer
+
+
+def load_model(model_path):
+    tokenizer = AutoTokenizer.from_pretrained(model_path)
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+    model = AutoModelForSequenceClassification.from_pretrained(model_path)
+    model.to(device)
+    return tokenizer, model, device
+
+
+def predict(text, tokenizer, model, device):
+    inputs = tokenizer(text, return_tensors="pt", padding=True, truncation=True)
+    inputs = {key: tensor.to(device) for key, tensor in inputs.items()}
+    outputs = model(**inputs)
+    logits = outputs.logits
+    predictions = torch.argmax(logits, dim=-1)
+    return predictions.item()
+
+
+def main():
+    model_path = valohai.inputs("model").path(process_archives=False)
+    if model_path.endswith(".zip"):
+        with zipfile.ZipFile(model_path, "r") as zip_ref:
+            zip_ref.extractall("/tmp/model")  # noqa: S108
```

Comment:
Please don't noqa the security lint, but use `tempfile.TemporaryDirectory()` instead.

This will also fix the other bug that's lurking here, namely blindly trusting `/tmp` to be a tempdir.

This repeats for all of this pattern throughout this PR.